website/docs/contributing/ADRs/back-end/POST-PUT-api-payload.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "ADR: POST/PUT API payload"
+---
+
+## Background
+
+Whenever we receive a payload in our backend for POST or PUT request we need to take into account backward compatibility. When we add a new field to an existing API payload, clients using the previous version of the payload will not know about that new field. This means that we need to make sure that the new field is optional. If we make the field required, clients using the previous version of the payload will result in overriding the value of the new field with an empty value or null.
+
+### Example: adding new setting field to project settings
+
+Project settings on unleash 5.3:
+```shell
+curl --location --request PUT 'http://localhost:4242/api/admin/projects/default' \
+--header 'Authorization: INSERT_API_KEY' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+  "id": "default",
+  "name": "Default",
+  "description": "Default project",
+  "defaultStickiness": "default",
+  "mode": "open"
+}'
+```
+
+New version of project settings (unleash 5.6):
+
+```shell
+curl --location --request PUT 'http://localhost:4242/api/admin/projects/default' \
+--header 'Authorization: INSERT_API_KEY' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+  "id": "default",
+  "name": "Default",
+  "description": "Default project",
+  "defaultStickiness": "default",
+  "featureLimit": 2
+}'
+```
+
+Pay attention to the new field feature limit. If a customer updates their server to 5.6 but their integration still does not send that field, it may result in the unwanted behavior of setting that field to empty in the database (in case the server assumes that not sending the field is tring to set it to empty).
```

Comment:
```suggestion
Pay attention to the new field feature limit. If a customer updates Unleash to 5.6 but their integration still does not send that field, it may result in the unwanted behavior of setting that field to empty in the database, in case the server assumes that not sending the field means setting it to empty / `null`.
```